inhibit/inhibit.go

```diff
@@ -50,33 +52,26 @@ func NewInhibitor(ap provider.Alerts, rs []*config.InhibitRule, mk types.Marker)
 	return ih
 }
 
-func (ih *Inhibitor) runGC() {
+func (ih *Inhibitor) runGC(ctx context.Context) {
 	for {
 		select {
 		case <-time.After(15 * time.Minute):
 			for _, r := range ih.rules {
 				r.gc()
 			}
-		case <-ih.stopc:
+		case <-ctx.Done():
 			return
```

Comment:
Think you should return `ctx.Err()`

Reply:
`ctx.Err()` just returns why a context was canceled or deadline exceeded. It doesn't actually tell us anything here.

```
        // If Done is not yet closed, Err returns nil.
        // If Done is closed, Err returns a non-nil error explaining why:
        // Canceled if the context was canceled
        // or DeadlineExceeded if the context's deadline passed.
        // After Err returns a non-nil error, successive calls to Err return the same error.
        Err() error
```

Reply:
true, but if non nil isn't worth logging?

Reply:
I don't think so, because we explicitly use the context to cancel the execution in the `Inhibitor.Stop` method.

Reply:
Gotcha 👍

Reply:
Yes, the error is always `!=` nil if the context was canceled. If we expect this to happen, there's generally no point in checking.